src/main/js/dashboard-container.js

```diff
@@ -2,21 +2,45 @@
  * Dashboard container to manage state and logic for the user dashboard page.
  *
  * @author Jean Dorancy
+ * @editor Xiaobing Hou
  */
 
 import React, { useEffect, useState } from "react";
 import Dashboard from "./dashboard";
 import AccountService from "./account/account-service";
+import TransactionService from "./tansaction/transaction-service";
 
 const DashboardContainer = (props) => {
-  const service = new AccountService();
+  const accountService = new AccountService();
+  const transactionService = new TransactionService();
   const [bankAccounts, setBankAccounts] = useState([]);
+  const [bankAccountId, setBankAccountId] = useState(0);
+  const [transactions, setTransactions] = useState([]);
+  const [transactionModalShow, setTransactionModalShow] = useState(false);
+  const [transactionConfModalShow, setTransactionConfModalShow] =
+    useState(false);
 
-  useEffect(() => {
-    service
+  /**
+   * Handle click on bank account, init the bank accounts list and transactions list
+   *
+   * @param bankAccountId
+   * @author Xiaobing Hou
+   */
+  const handleBankAccounts = (bankAccountId) => {
+    accountService
       .findAllBankAccount()
       .then(function (response) {
         setBankAccounts(response.data.additionalData || []);
+        if (bankAccountId) {
+          setBankAccountId(bankAccountId);
+          handleTransactions(setTransactions, bankAccountId);
+        } else {
+          setBankAccountId(response.data.additionalData[0].id);
+          handleTransactions(
+            setTransactions,
+            response.data.additionalData[0].id
+          );
```

Comment:
The transaction table can hold transactions from any of the accounts which belongs to the user not just one hence each transaction will need to keep the bank account id which he belongs to.

Reply:
![image](https://user-images.githubusercontent.com/53858277/172072831-f05d0564-d42d-4249-bb9c-040b3970f425.png)
My design is to switch the transaction page by clicking on the account card.
I think it will be clearer

I have done some change and commited to make it better